### PR TITLE
Precommit docs

### PR DESCRIPTION
**Pull request by altusgerona**
## Types of changes
Please put an `x` in the box that applies

- [ ] **Bugfix**
- [ ] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [x] **Documentation Update**
- [ ] **Other (please describe)**

## Description
Add instructions for pre-commit integration in the README.md itself. 

## Checklist:
- [ ] Add test cases to all the changes you introduce
- [x] Run `./script/lint` and `./script/test` locally to ensure this change passes linter check and test
- [x] Test the changes on the local machine manually
- [x] Update the documentation for the changes

## Related Issue
#196  #209

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -74,6 +74,20 @@ or the shortcut
...
 ```
 
+### Integrating with Pre-commit
+Commitizen can lint your commit message for you with `cz check`.
+You can integrate this in your [pre-commit](https://pre-commit.com/) config with:
+```yaml
+---
+repos:
+  - repo: https://github.com/commitizen-tools/commitizen
+    rev: v1.17.0
```
Comment (Lee-W) [inline docs/README.md:84]: I would suggest updating the version to the latest commitizen

File: docs/README.md [hunk 2]
```diff
@@ -74,6 +74,20 @@ or the shortcut
 cz c
 ```
 
+### Integrating with Pre-commit
+Commitizen can lint your commit message for you with `cz check`.
+You can integrate this in your [pre-commit](https://pre-commit.com/) config with:
+```yaml
```
Comment (Lee-W) [inline docs/README.md:80]: Adding this config file is not enough for `pre-commit`, I'd suggest add the `pre-commit install --hook-type commit-msg` command

File: docs/README.md (hunk 1 above)
Comment (altusgerona) [reply docs/README.md:84]: Wouldn't it be better if we just leave it at `master`. That's actually what I use for our company projects.

File: docs/README.md (hunk 2 above)
Comment (altusgerona) [reply docs/README.md:80]: Hmm. This already assumes that the dev has run `pre-commit` before tho. When I was trying this out for myself I just added that snippet to my config and reran pre-commit. I think the `stages: [commit-msg]` is enough.

File: docs/README.md (hunk 1 above)
Comment (Lee-W) [reply docs/README.md:84]: My reason for not doing so is that pre-commit caches repos. You might not notice that you're still using the older version.

Comment (altusgerona) [reply docs/README.md:84]: True, but it's the same with using a fixed version. Also, this is the README. We'd have to update the version every time we bump.

Comment (altusgerona) [reply docs/README.md]: I think we'll be documenting `pre-commit` for them with that. It's already documented by pre-commit themselves. But if you guys want I'll put it in. Just let me know.

**Comment by Woile** [comment]
Can't we add to pyproject.toml

```
version_files = [
  "pyproject.toml:version",
  "commitizen/__version__.py",
  "docs/README.md:\trev:"
]
```

or something like that?

**Comment by altusgerona** [comment]
> Can't we add to pyproject.toml
> 
> ```
> version_files = [
>   "pyproject.toml:version",
>   "commitizen/__version__.py",
>   "docs/README.md:\trev:"
> ]
> ```
> 
> or something like that?
Sure. But I think it'll just make it more verbose than it has to be. Other python libs that use pre-commit—pre-commit included just use `master` or `stable`.

**Comment by Woile** [comment]
`master` is fine by me as well, I have no issues with it

File: docs/README.md (hunk 2 above)
Comment (altusgerona) [reply docs/README.md:80]: Hey @Lee-W, just to follow up, you're right. It needs the `pre-commit install --hook-type commit-msg` command.

Comment (Lee-W) [reply docs/README.md:80]: Ah, I miss this part when I merged this PR.  No worries. I'll add it in another PR.

**Pull request merged**